test/scripts/email-config.js

```diff
@@ -0,0 +1,224 @@
+        })
+    })
+
+    describe('write config without regex:', () => {
+      beforeEach(() => {
+        config.socketlabs = {
+          percentage: 10
+        }
+        return cp.execAsync(`echo '${JSON.stringify(config)}' | node scripts/email-config write`, { cwd })
+      })
+
+      it('read prints the config to stdout', () => {
+        return cp.execAsync('node scripts/email-config read', { cwd })
+          .then((stdout, stderr) => {
+            assert.equal(stdout, `${JSON.stringify(config, null, '  ')}\n`)
+            assert.equal(stderr, undefined)
+          })
+      })
+
+      it('check matching email prints the both configs to stdout', () => {
+        return cp.execAsync('node scripts/email-config check foo@example.com', { cwd })
+          .then((stdout, stderr) => {
+            assert.equal(stdout, `${JSON.stringify(config, null, '  ')}\n`)
+            assert.equal(stderr, undefined)
+          })
+      })
+
+      it('check non-matching email prints one config to stdout', () => {
+        return cp.execAsync('node scripts/email-config check bar@example.com', { cwd })
+          .then((stdout, stderr) => {
+            const expected = {
+              socketlabs: config.socketlabs
+            }
+            assert.equal(stdout, `${JSON.stringify(expected, null, '  ')}\n`)
+            assert.equal(stderr, undefined)
+          })
+      })
+
+      describe('revert:', () => {
+        beforeEach(() => {
+          return cp.execAsync('node scripts/email-config revert', { cwd })
+        })
+
+        it('read prints the previous config to stdout', () => {
+          return cp.execAsync('node scripts/email-config read', { cwd })
+            .then((stdout, stderr) => {
+              const expected = {
+                sendgrid: config.sendgrid
+              }
+              assert.equal(stdout, `${JSON.stringify(expected, null, '  ')}\n`)
+              assert.equal(stderr, undefined)
+            })
+        })
+
+        it('check matching email prints the previous config to stdout', () => {
+          return cp.execAsync('node scripts/email-config check foo@example.com', { cwd })
+            .then((stdout, stderr) => {
+              const expected = {
+                sendgrid: config.sendgrid
+              }
+              assert.equal(stdout, `${JSON.stringify(expected, null, '  ')}\n`)
+              assert.equal(stderr, undefined)
+            })
+        })
+
+        it('check non-matching email does not print', () => {
+          return cp.execAsync('node scripts/email-config check bar@example.com', { cwd })
+            .then((stdout, stderr) => {
+              assert.equal(stdout, '{}\n')
+              assert.equal(stderr, undefined)
+            })
+        })
+      })
+    })
+
+    describe('revert:', () => {
+      beforeEach(() => {
+        return cp.execAsync('node scripts/email-config revert', { cwd })
+      })
+
+      it('read does not print', () => {
```

Comment:
Ah I see, this is expected.

Reply:
Yeah, the intention is to be UNIX/command-line friendly. So the only thing that goes to `stdout` is the config, if there is any. Otherwise it risks having people accidentally pipe status messages to some other file/command.